addons/hr_holidays/static/src/persona_model_patch.js

```diff
@@ -0,0 +1,21 @@
+/** @odoo-module */
+
+import { Persona } from "@mail/core/common/persona_model";
+import { patch } from "@web/core/utils/patch";
+
+patch(Persona.prototype, {
+    /**
+     * @overwrite
+     */
+    updateImStatus(newStatus) {
+        if (newStatus == "online" && this.out_of_office_date_end) {
+            this.im_status = "leave_online";
+        } else if (newStatus == "offline" && this.out_of_office_date_end) {
+            this.im_status = "leave_offline";
+        } else if (newStatus == "away" && this.out_of_office_date_end) {
+            this.im_status = "leave_away";
+        } else {
+            this.im_status = newStatus;
+        }
```

Comment:
`else` should be super call rather than copy-paste original code: `return super.updateImStatus(...arguments);`